docs/features/search/declarative-integration.md

```diff
@@ -0,0 +1,230 @@
+
+_Example disabling the search page extension_
+
+```yaml
+# app-config.yaml
+app:
+  experimental:
+    packages: 'all' # enable packages auto discovery
+  extensions:
+    - plugin.search.page: false # ✨
+```
+
+_Example setting the search sidebar item label_
+
+```yaml
+# app-config.yaml
+app:
+  experimental:
+    packages: 'all' # enable packages auto discovery
+  extensions:
+    - plugin.search.nav.index: # ✨
+        config:
+          label: 'Search Page'
+```
+
+> **Known limitations:**
+> It is currently not possible to open modals in sidebar items and also configure a different icon via configuration file, but it is already on the maintainers' radar.
+
+### Customizations
+
+Plugin developers can use the `createSearchResultItemExtension` factory provided by the `@backstage/plugin-search-react` for building their own custom `Search` result item extensions.
+
+_Example creating a custom `TechDocsSearchResultItemExtension`_
+
+```tsx
+// plugins/techdocs/alpha.tsx
+import { createSearchResultListItemExtension } from '@backstage/plugin-search-react/alpha';
+
+/** @alpha */
+export const TechDocsSearchResultListItemExtension =
+  createSearchResultListItemExtension({
+    id: 'techdocs',
+    configSchema: createSchemaFromZod(z =>
+      z.object({
+        noTrack: z.boolean().default(false),
+        lineClamp: z.number().default(5),
+      }),
+    ),
+    predicate: result => result.type === 'techdocs',
+    component: async ({ config }) => {
+      const { TechDocsSearchResultListItem } = await import(
+        './components/TechDocsSearchResultListItem'
+      );
+      return props => <TechDocsSearchResultListItem {...props} {...config} />;
+    },
+  });
+```
+
+In the snippet above, a plugin developer is providing a custom component for rendering search results of type "techdocs". The custom result item extension will be enabled by default once the `@backstage/plugin-techdocs` package is installed, that means adopters don't have to enable the extension manually via configuration file.
+
+When a Backstage adopter doesn't want to use the custom `TechDocs` search result item after installing the `TechDocs` plugin, they could disable it via Backstage configuration file:
+
+```yaml
+# app-config.yaml
+app:
+  experimental:
+    packages: 'all'
+  extensions:
+    - plugin.search.result.item.techdocs: false # ✨
+```
+
+Because a configuration schema was provided to the extension factory, Backstage adopters will be able to customize `TechDocs` search results **line clamp** that defaults to 3 and also **disable automatic analytics events tracking**:
+
+```yaml
+# app-config.yaml
+app:
+  experimental:
+    packages: 'all' # enable packages auto discovery
+  extensions:
+    - plugin.search.result.item.techdocs:
+        config: # ✨
+          noTrack: true
+          lineClamp: 3
+```
+
+The `createSearchResultItemExtension` function returns a Backstage's extension representation as follows:
```

Comment:
I'd say this is too much in the weeds for plugin-specific docs, but also a good explanation so I think we should keep it. Maybe a TODO to move this explanation to a more central place in the future.

Reply:
Good idea, comment added 🙌🏻 : https://github.com/backstage/backstage/pull/20328/commits/8623fdeca87e79e4b1769db84c4482032dbbda9c